compiler/AST/LoopExpr.cpp

```diff
@@ -198,7 +198,11 @@ bool LowerLoopExprVisitor::enterLoopExpr(LoopExpr* node) {
   if (node->getStmtExpr() == NULL) {
     // Don't touch LoopExprs in DefExprs, they should be copied later into
     // BlockStmts.
-    INT_ASSERT(isDefExpr(node->parentExpr));
+
+    // While this works for correct codes, it results in assertion errors
+    // for incorrect codes that don't generate errors until resolution:
+    //
+    //    INT_ASSERT(isDefExpr(node->parentExpr));
```

Comment:
Could you check that the failing case runs OK when the compiler is run under valgrind and verify?

Reply:
For the new test?  Good thought!

- [x] `--compopts --verify` works
- [x] `--valgrind` works
- [x] `export CHPL_SANITIZE=address` works